src/de/impls.rs

```diff
@@ -446,28 +446,26 @@ where
     fn decode<D: Decoder>(decoder: &mut D) -> Result<Self, DecodeError> {
         decoder.claim_bytes_read(core::mem::size_of::<[T; N]>())?;
 
-        // TODO: we can't limit `T: 'static` because that would break other things
-        // but we want to have this optimization
-        // This will be another contendor for specialization implementation
-        // if TypeId::of::<u8>() == TypeId::of::<T>() {
-        //     let mut buf = [0u8; N];
-        //     decoder.reader().read(&mut buf)?;
-        //     let ptr = &mut buf as *mut _ as *mut [T; N];
-
-        //     // Safety: we know that T is a u8, so it is perfectly safe to
-        //     // translate an array of u8 into an array of T
-        //     let res = unsafe { ptr.read() };
-        //     Ok(res)
-        // }
-        let result = super::impl_core::collect_into_array(&mut (0..N).map(|_| {
-            // See the documentation on `unclaim_bytes_read` as to why we're doing this here
-            decoder.unclaim_bytes_read(core::mem::size_of::<T>());
-            T::decode(decoder)
-        }));
-
-        // result is only None if N does not match the values of `(0..N)`, which it always should
-        // So this unwrap should never occur
-        result.unwrap()
+        if unty::type_equal::<T, u8>() {
+            let mut buf = [0u8; N];
+            decoder.reader().read(&mut buf)?;
+            let ptr = &mut buf as *mut _ as *mut [T; N];
+
+            // Safety: we know that T is a u8, so it is perfectly safe to
```

Comment:
The documentation of `unty` does not support this statement, but I think it's a documentation bug in unty. The documentation for `type_equal` says:

> _"Note that this may give false positives if any of the types have lifetimes."_

So, since `T` might contain lifetimes here, according to the documentation "T is u8" might be a false positive, so `T` might not be `u8` here.

Probably it needs to say the false positive is when both of the types have lifetimes.

Reply:
Updated documentation in unty 0.0.3, is this good now?

Reply:
:+1:

Consider also updating `impl Encode for [T; N]`, and the 3 `VecDeque` impls.